src/js/components/DataTable/doc.js

```diff
@@ -259,6 +265,16 @@ export const doc = DataTable => {
       object. If 'onSelect' is provided, the CheckBoxes are enabled
       and this function can be used to track select changes.`,
     ),
+    show: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.shape({ page: PropTypes.number }),
+    ])
+      .description(
+        `To be used in conjunction with paginate. If provided as a number, 
```

Comment:
I'm not sure we need to make this only for when using `paginate`. Even with InfiniteScroll it should be able to work.

Reply:
Agreed. Adjusted docs and made sure functionality is working.